crypto/ec/ec_key.c

```diff
@@ -119,9 +119,11 @@ EC_KEY *EC_KEY_copy(EC_KEY *dest, const EC_KEY *src)
                     return NULL;
             }
             if (!BN_copy(dest->priv_key, src->priv_key))
+                BN_free(dest->priv_key);
```

Comment:
EC_KEY_copy is a public API function. It is part of the contract of the API that an application using it passes in two references to EC_KEY objects that the application already owns. This function does not transfer any ownership of that memory, so after it completes (even if it fails) then ownership still lies with the calling application - which implies that it is the calling application's responsibility to free those EC_KEY objects when they are finished with. If this does not occur then it is a bug in the calling application - not in the implementation of this function.